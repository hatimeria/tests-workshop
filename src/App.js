import { useState } from 'react';
import './App.css';

function App() {
  const [name, setName] = useState('stranger');
  const [error, setError] = useState('');
  const handleChange = value => {
    !/[^a-zA-Z]/.test(value) ? setName(value) : setError('You cannot be named like this!');
  }
  return (
    <main>
      <h1>Hello {name}!</h1>
      {error && <p role="alert"></p>}
      <form>
        <label htmlFor="name">Your name:</label>
        <input type="text" id="name" onChange={e => handleChange(e.target.value)} />
      </form>
    </main>
  );
}

export default App;
