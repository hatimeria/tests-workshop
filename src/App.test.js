import { render, screen, fireEvent } from '@testing-library/react';
import App from './App';

test('renders welcome message', () => {
  render(<App />);
  const message = screen.getByText('Hello stranger!');
  expect(message).toBeInTheDocument();
});

test('renders input and button', () => {
  render(<App />);
  const input = screen.getByRole('textbox');
  expect(input.value).toBe('');
});

test('display name after sending form', () => {
  render(<App />);
  const input = screen.getByRole('textbox');
  const message = screen.getByText('Hello stranger!');
  fireEvent.change(input, { target: { value: 'Julia' } });
  expect(message.textContent).toBe('Hello Julia!');
});