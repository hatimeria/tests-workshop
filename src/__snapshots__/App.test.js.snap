// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<main>
  <h1>
    Hello 
    stranger
    !
  </h1>
  
  <form>
    <label
      htmlFor="name"
    >
      Your name:
    </label>
    <input
      id="name"
      onChange={[Function]}
      type="text"
    />
  </form>
  <div>
    
    
  </div>
</main>
`;
